backend/src/services/attendances/attendances.service.js

```diff
@@ -266,6 +271,286 @@ class AttendancesService {
 
     return await this.attendancesRepo.getAttendanceStats(lecture.id, course.id);
   }
+
+  /**
+   * Create attendance by code.
+   * Finds the lecture by code and creates attendance for the current user.
+   * This is the simplified flow for students who only have a code.
+   *
+   * @param {Object} user - Current user object
+   * @param {Object} course - Course object (from req.course)
+   * @param {Object|null} enrollment - Enrollment object (from req.enrollment, null if not enrolled)
+   * @param {string} code - Attendance code
+   * @returns {Promise<Object>} Created attendance object
+   */
+  async createAttendanceByCode(user, course, enrollment, code) {
+    if (!this.lecturesRepo) {
+      const error = new Error('Lectures repository not available');
+      error.code = 'INTERNAL_SERVER_ERROR';
+      throw error;
+    }
+
+    // Find lecture by code (only active codes)
+    const lecture = await this.lecturesRepo.getLectureByCode(course.id, code);
+    if (!lecture) {
+      // Check if code exists but expired for more specific error message
+      const expiredLecture = await this.lecturesRepo.getLectureByCodeAnyStatus(
+        course.id,
+        code
```

Comment:
[Suggestion]
Rather than making 2 db calls, fetch in one call (get lecture by any status) and then determine based on returned record which error to return.